src/nodes.coffee

```diff
@@ -4732,12 +4735,11 @@ exports.SwitchWhen = class SwitchWhen extends Base
 # Single-expression **Ifs** are compiled into conditional operators if possible,
 # because ternaries are already proper expressions, and don’t need conversion.
 exports.If = class If extends Base
-  constructor: (condition, @body, options = {}) ->
+  constructor: (@condition, @body, options = {}) ->
     super()
-    @condition = if options.type is 'unless' then condition.invert() else condition
```

Comment:
We have to defer inverting `@condition` so that the original condition is still intact at AST generation time